backend/Dockerfile

```
@@ -17,23 +17,39 @@ COPY pyproject.toml uv.lock ./
+WORKDIR /app
+COPY backend/app ./app
+RUN python -m compileall -q -x '(^|/)(tests|test_).*' app
+
 # Runtime stage
-FROM python:3.12-slim
+FROM python:3.12-slim AS runtime
 
 WORKDIR /app
 
+# Install runtime dependencies: curl (healthcheck), ffmpeg (yt-dlp video processing)
+RUN apt-get update && apt-get install -y --no-install-recommends \
```

Hadolint (2.14.0)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)